solutions/cf-616D.mdx

```diff
@@ -0,0 +1,79 @@
+---
+id: cf-616D
+source: CF
+title: Longest k-Good Segment
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/22712)
+
+<LanguageSection>
+
+<JavaSection>
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+## Java Implementation
+
+```java
+import java.util.*;
+public class KGood {
+	CodeSnip{Kattio}
+
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio();
+		int N = io.nextInt();
+		int K = io.nextInt();
+		int[] freq = new int[MAX];  // Frequency of each number in the window.
```

Comment:
does this compile? what is `MAX`

Reply:
it's a constant, i told her to change it to something that's dynamically calculated, dw about it

Reply:
I forgot to delete it from earlier, it's gone now.